snuba/migrations/runner.py

```diff
@@ -66,3 +110,29 @@ def _mark_completed(self, group: MigrationGroup, migration_id: str) -> None:
             ClickhouseClientSettings.MIGRATE
         )
         connection.execute(statement, data)
+
+        if self.__migrations_status is not None:
+            self.__migrations_status[migration_key] = status
```

Comment:
When can this be None ? 
I think lazily initializing `self.__migration_status` makes this logic a bit confusing since it is harder to understand without going back and forth whether in a given state the mapping can be none. 
Why not just making the run_all method more stateless:
1) load all migrations from DB and keep the status in a 
2) for each on of them, run it. No need to update the migration_status variable

Reply:
Caching the results isn't really useful right now. I thought it might turn out to be better later to not need to keep querying for them, but I'm not sure if we'll even end up needing that so I've removed it.